### ISSUE TITLE
Make error

### ISSUE THREAD

**Issue by gldpittolzinho** [p4lang/behavioral-model#453]
When I try to run make on the code I get an error:
```
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../.. -DBMLOG_DEBUG_ON -DBMLOG_TRACE_ON -DBMELOG_ON -DBMTHRIFT_ON -DBMNANOMSG_ON -I../../include -isystem../../third_party/jsoncpp/include -isystem../../third_party/spdlog -I./../BMI -I./../bf_lpm_trie -pthread -Wall -Werror -Wextra -g -O2 -MT deparser.lo -MD -MP -MF .deps/deparser.Tpo -c deparser.cpp  -fPIC -DPIC -o .libs/deparser.o
In file included from ../../include/bm/bm_sim/logger.h:36:0,
                 from deparser.cpp:23:
../../third_party/spdlog/bm/spdlog/spdlog.h:71:133: error: ‘function’ in namespace ‘std’ does not name a template type
 cy overflow_policy = async_overflow_policy::block_retry, const std::function<void()>& worker_warmup_cb = nullptr, const std::chrono::milliseconds& flush_interval_ms = std::chrono::milliseconds::zero());
                                                                     ^~~~~~~~
../../third_party/spdlog/bm/spdlog/spdlog.h:71:141: error: expected ‘,’ or ‘...’ before ‘<’ token
 low_policy = async_overflow_policy::block_retry, const std::function<void()>& worker_warmup_cb = nullptr, const std::chrono::milliseconds& flush_interval_ms = std::chrono::milliseconds::zero());
                                                                     ^
In file included from ../../include/bm/bm_sim/logger.h:36:0,
                 from deparser.cpp:23:
../../third_party/spdlog/bm/spdlog/spdlog.h:71:6: error: default argument missing for parameter 3 of ‘void spdlog::set_async_mode(size_t, spdlog::async_overflow_policy, int)’
 void set_async_mode(size_t queue_size, const async_overflow_policy overflow_policy = async_overflow_policy::block_retry, const std::function<void()>& worker_warmup_cb = nullptr, const std::chrono::milliseconds& flush_interval_ms = std::chrono::milliseconds::zero());
      ^~~~~~~~~~~~~~
In file included from ../../third_party/spdlog/bm/spdlog/spdlog.h:155:0,
                 from ../../include/bm/bm_sim/logger.h:36,
                 from deparser.cpp:23:
../../third_party/spdlog/bm/spdlog/details/spdlog_impl.h:140:197: error: ‘void spdlog::set_async_mode(size_t, spdlog::async_overflow_policy, const std::function<void()>&, const milliseconds&)’ should have been declared inside ‘spdlog’
 <void()>& worker_warmup_cb, const std::chrono::milliseconds& flush_interval_ms)
                                                                               ^
Makefile:535: recipe for target 'deparser.lo' failed
make[3]: *** [deparser.lo] Error 1
make[3]: Leaving directory '/home/l/p4c/behavioral-model/src/bm_sim'
Makefile:541: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/home/l/p4c/behavioral-model/src'
Makefile:450: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/l/p4c/behavioral-model'
Makefile:380: recipe for target 'all' failed
make: *** [all] Error 2
$ 

```
I've run install_dependencies.sh successfully.

```
$ thrift -version
Thrift version 0.9.2
$ ldconfig -p | grep thrift
	libthriftz-0.9.2.so (libc6,x86-64) => /usr/local/lib/libthriftz-0.9.2.so
	libthriftnb-0.9.2.so (libc6,x86-64) => /usr/local/lib/libthriftnb-0.9.2.so
	libthrift-0.9.2.so (libc6,x86-64) => /usr/local/lib/libthrift-0.9.2.so
$ 

```

**Comment by antoninbas** [comment]
Should be fixed by #451 
Please include your compiler version here. Is it gcc7? We may need to add a Travis regression with gcc7 to avoid future issues like this one.

**Comment by gldpittolzinho** [comment]
it is gcc 7.2.0. thanks for the quick reply by the way :)
